fe/fe-core/src/main/java/org/apache/doris/analysis/Analyzer.java

```diff
@@ -759,6 +765,18 @@ public TupleDescriptor registerOlapTable(Table table, TableName tableName, List<
             slot.setIsNullable(col.isAllowNull());
             String key = tableRef.aliases[0] + "." + col.getName();
             slotRefMap.put(key, slot);
+
+            if (col.getType().isVariantType()) {
+                LOG.debug("add subColumnSlotRefMap, key {}, column {}", key, col.toThrift());
+                subColumnSlotRefMap.put(key, Maps.newTreeMap(
```

Comment:
Is this duplicated code necessary since it can be handled by registerColumnRef?

Reply:
yes, i will refactor it